### PR TITLE
test(#1902): Implement unit tests for pipeline element 'TextFilterProcessor'

### LINKED ISSUE #1902: Implement unit tests for pipeline element 'TextFilterProcessor'
This issue is about implementing unit tests for our pipeline element [TextFilterProcessor](https://github.com/apache/streampipes/blob/752289c6acfc8fa10398c920d678c5c02143ed5c/streampipes-extensions/streampipes-processors-filters-jvm/src/main/java/org/apache/streampipes/processors/filters/jvm/processor/textfilter).
An example of how to write unit tests for pipeline elements can be found [here](https://github.com/apache/streampipes/tree/ca4e0c2d44c1a713caaee3e9c1bff46096f41c53/streampipes-extensions/streampipes-processors-transformation-jvm/src/test/java/org/apache/streampipes/processors/transformation/jvm/processor).

### Mentoring
As this issue is marked as a good first issue and for the `hacktoberfest` event: one of @dominikriemer, @tenthe, @RobertIndie, or @bossenti are happy to provide help for getting started, just tag (one of) them if you want to start working on this issue and need some help.

### PR DESCRIPTION
**Pull request by Aditi840**
### Purpose

closes #1902 

### Remarks

PR introduces (a) breaking change(s): no

PR introduces (a) deprecation(s): no

### CODE REVIEW

**Comment by Aditi840** [comment]
@bossenti, will you check my pull request

File: streampipes-extensions/streampipes-processors-transformation-jvm/src/test/java/org/apache/streampipes/processors/transformation/jvm/Additional Tests [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        assertNull(processor.getKeyword());
+        assertNull(processor.getStringOperator());
+        assertNull(processor.getFilterProperty());
+    }
+
+    @Test
+    public void testModelDeclaration() {
+        TextFilterProcessor processor = new TextFilterProcessor();
+        assertNotNull(processor.declareModel());
+        // Add more assertions to validate the description
```
Comment (bossenti) [inline streampipes-extensions/streampipes-processors-transformation-jvm/src/test/java/org/apache/streampipes/processors/transformation/jvm/Additional Tests:42]: Are you willing to come up with some here?

File: streampipes-extensions/streampipes-processors-transformation-jvm/src/test/java/org/apache/streampipes/processors/transformation/jvm/Additional Tests [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+        processor.onEvent(nonMatchingEvent, spOutputCollector);
+        assertEquals(0, spOutputCollector.getCollectedEvents().size());
+    }
+
+    @Test
+    public void testOnDetach() {
+        TextFilterProcessor processor = new TextFilterProcessor();
+        assertDoesNotThrow(() -> processor.onDetach());
+        // Add assertions if there is any specific behavior to test
```
Comment (bossenti) [inline streampipes-extensions/streampipes-processors-transformation-jvm/src/test/java/org/apache/streampipes/processors/transformation/jvm/Additional Tests:88]: I think there is no more to test here, so you can safely remove this comment

File: streampipes-extensions/streampipes-processors-transformation-jvm/src/test/java/org/apache/streampipes/processors/transformation/jvm/Additional Tests
Comment (bossenti) [inline streampipes-extensions/streampipes-processors-transformation-jvm/src/test/java/org/apache/streampipes/processors/transformation/jvm/Additional Tests:1]: This is not a valid `java` class. Can you please give it a valid file name and put it in the corresponding package?

File: streampipes-extensions/streampipes-processors-transformation-jvm/src/test/java/org/apache/streampipes/processors/transformation/jvm/Additional Tests [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    @Test
+    public void testOnEvent() {
+        TextFilterProcessor processor = new TextFilterProcessor();
+        when(processorParams.extractor().singleValueParameter(anyString(), eq(String.class))).thenReturn("keyword");
+        when(processorParams.extractor().selectedSingleValue(anyString(), eq(String.class))).thenReturn("MATCHES");
+        when(processorParams.extractor().mappingPropertyValue(anyString())).thenReturn("text");
+
+        processor.onInvocation(processorParams, spOutputCollector, eventProcessorRuntimeContext);
+
+        Event matchingEvent = createMockEvent("Some text that MATCHES the keyword");
```
Comment (bossenti) [inline streampipes-extensions/streampipes-processors-transformation-jvm/src/test/java/org/apache/streampipes/processors/transformation/jvm/Additional Tests:68]: This text only contains the keyword, doesn't it?
If you would like to test for equality, your string should just be `keyword`.

File: streampipes-extensions/streampipes-processors-transformation-jvm/src/test/java/org/apache/streampipes/processors/transformation/jvm/Additional Tests [hunk 4]
```diff
@@ -0,0 +1,96 @@
+import org.apache.streampipes.commons.exceptions.SpRuntimeException;
```
Comment (bossenti) [inline streampipes-extensions/streampipes-processors-transformation-jvm/src/test/java/org/apache/streampipes/processors/transformation/jvm/Additional Tests:1]: This file misses our Apache license header, see an example [here](https://github.com/apache/streampipes/blob/ca4e0c2d44c1a713caaee3e9c1bff46096f41c53/streampipes-extensions/streampipes-processors-transformation-jvm/src/test/java/org/apache/streampipes/processors/transformation/jvm/processor/value/change/TestChangedValueDetectionProcessor.java#L1).

**Review by bossenti — COMMENTED** [review]
thanks a lot for this PR @Aditi840 🙏🏼 
I think the test cases are quite extensive, really appreciate this.
However, there are a few things to adapt before we can merge the PR.

**Comment by Aditi840** [comment]
@bossenti, I have made some changes to the file will you check them.

**Comment by dominikriemer** [comment]
@Aditi840 I had a look at your test and want to give some feedback:
* The test class is intended for the `TextFilterProcessor` class, but your code added is in a module that doesn't contain a dependency on the `processors-filters-jvm` module which provides the class to test.
* The code you provided uses imports that cannot be resolved due to missing dependencies.
* The code uses invented methods which are not present in our codebase, such as `resetCollectedEvents` from the `SpOutputCollector` class or a `ModelDeclaration` class which doesn't even exist.

We are a friendly and open community and are willing to help non-experienced coders, but my general advise is that you don't provide contributions to open source projects that are based on ChatGPT and which are completely untested and wrong. Note that many people who provide real code to open source projects do that in their spare time. Reviewing PRs which come out to be mostly nonsense is a waste of time and just not nice.

Having said that, I'll close this PR.

**Pull request closed without merging**